src/Features/Core/Portable/EmbeddedLanguages/DateAndTime/EmbeddedCompletionContext.cs

```diff
@@ -33,31 +52,91 @@ internal partial class DateAndTimeEmbeddedCompletionProvider
                 }
 
                 _replacementSpan = TextSpan.FromBounds(startPosition, context.Position);
+
+                (_userFormatPrefix, _userFormatSuffix) = GetUserFormatParts(virtualChars, startPosition, context.Position);
             }
 
-            private static DateTime s_exampleDateTime = DateTime.Parse("2009-06-15T13:45:30.1234567Z");
-            private static CultureInfo s_enUsCulture = CultureInfo.GetCultureInfo("en-US");
-            private static CultureInfo s_primaryCulture = CultureInfo.CurrentCulture;
-            private static CultureInfo? s_secondaryCulture = s_primaryCulture.Equals(s_enUsCulture) ? null : s_enUsCulture;
+            private static (string prefix, string suffix) GetUserFormatParts(
+                VirtualCharSequence virtualChars, int startPosition, int endPosition)
+            {
+                virtualChars = virtualChars.IsDefault ? VirtualCharSequence.Empty : virtualChars;
```

Comment:
If this was a default case because it failed to parse, should the prefix/suffix also be null so it's more explicit those may not exist?

Reply:
i liked that this just fell out as you simply get an empty prefix/suffix (and the rest of the code doesn't need to care).  Seems like it would be adding a null-state, and then having to check that null-state later.  I'd prefer not :)